cmd/rocsp-tool/client.go

```diff
@@ -57,11 +58,25 @@ func (cl *client) loadFromDB(ctx context.Context, speed ProcessingSpeed) error {
 		return fmt.Errorf("no entries in certificateStatus (where notAfter >= %s)", startTime)
 	}
 
+	// Find the current maximum id in certificateStatus. We do this because the table is always
+	// growing. If we scanned until we saw a batch with no rows, we would scan forever.
+	var maxID *int64
+	err = cl.db.QueryRowContext(
+		ctx,
+		"SELECT MAX(id) FROM certificateStatus",
+	).Scan(&maxID)
```

Comment:
Do we actually need to pass `&maxID` here, or is just passing `maxID` (because it is already a pointer type) sufficient? It seems like it still won't instantiate anything if it gets no results, and `maxID` will still be `nil`.

(Same for `minID` above, although I didn't think of this question when reviewing this code the first time.)

Reply:
We can't pass just `maxID` because it doesn't point at any memory in which to store the result. We could set it up so that it does - `maxID := new(int64)`, but then we wouldn't have a clean way to check for "no rows." (We could check for 0, but that's a potentially valid value for ID).